clients/src/test/java/org/apache/kafka/common/network/SelectorTest.java

```diff
@@ -580,6 +583,49 @@ public void testConnectDisconnectDuringInSinglePoll() throws Exception {
         control.verify();
     }
 
+    @Test
+    public void testOutboundConnectionsCountInConnectionCreationMetric() throws Exception {
+        // create connections
+        int conns = 5;
+        InetSocketAddress addr = new InetSocketAddress("localhost", server.port);
+        for (int i = 0; i < conns; i++)
+            connect(Integer.toString(i), addr);
+
+        // Poll continuously, as we cannot guarantee that the first call will see all connections
+        for (int i = 0; i < 10; i++) {
+            selector.poll(0L);
+            if (selector.connected().size() == conns)
+                break;
+
+            Thread.sleep(100);
```

Comment:
You could just do `selector.poll(100)` instead of `poll(0) + sleep(100)`. `poll()` returns when an operation is ready, so we are not waiting unnecessarily.